go/couler/conversion/argo_workflow.go

```diff
@@ -7,58 +7,88 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-const sequentialStepsTemplateSuffix = "main-template"
+const entrypointTemplateSuffix = "main-template"
 
 // ConvertToArgoWorkflow converts a workflow from protobuf to an Argo Workflow
 func ConvertToArgoWorkflow(workflowPb *pb.Workflow, namePrefix string) (wfv1.Workflow, error) {
-	templates := []wfv1.Template{{Name: namePrefix + sequentialStepsTemplateSuffix}}
-	// TODO: Handle DAG tasks.
+	// Check whether the workflow is DAG.
+	dagMode := false
 	for _, step := range workflowPb.GetSteps() {
-		seqStep := step.Steps[0]
-		templates[0].Steps = append(templates[0].Steps,
-			wfv1.ParallelSteps{
-				Steps: []wfv1.WorkflowStep{{Name: seqStep.TmplName, Template: seqStep.TmplName}}})
-		template := wfv1.Template{Name: seqStep.TmplName}
-		// TODO: Check mutual exclusivity of different specs.
-		if seqStep.GetContainerSpec() != nil || seqStep.GetScript() != "" {
-			containerSpec := seqStep.GetContainerSpec()
-			container := &corev1.Container{
-				Image:   containerSpec.GetImage(),
-				Command: containerSpec.GetCommand(),
-				// TODO: Convert type map[string]*any.Any) to type []EnvVar that's supported by Argo.
-				//Env: containerSpec.GetEnv(),
-			}
-			if script := seqStep.GetScript(); script != "" {
-				template.Script = &wfv1.ScriptTemplate{
-					Container: *container,
-					Source:    script,
-				}
-			} else {
-				template.Container = container
-			}
-		} else if resourceSpec := seqStep.GetResourceSpec(); resourceSpec != nil {
-			template.Resource = &wfv1.ResourceTemplate{
-				// TODO: Check whether these hard-coded fields need to be exposed.
-				SetOwnerReference: true,
-				Action:            "create",
-				Manifest:          resourceSpec.GetManifest(),
-				SuccessCondition:  resourceSpec.GetSuccessCondition(),
-				FailureCondition:  resourceSpec.GetFailureCondition(),
-			}
+		if step.Steps[0].Dependencies != nil {
+			dagMode = true
+			break
+		}
+	}
+	templates := []wfv1.Template{{Name: namePrefix + entrypointTemplateSuffix}}
+	// Convert steps to DAG tasks.
+	if dagMode {
+		templates[0].DAG = &wfv1.DAGTemplate{
```

Comment:
Can have two functions constructing workflow templates to make the function `ConvertToArgoWorkflow` shorter.

Reply:
Thanks. Updated the PR.